Db.php

```diff
@@ -3,126 +3,205 @@
-        $info = self::getConnectionInfo($collection, $read_only);
-        list($mongo, $db_name) = $info;
-        if (!$mongo->connected) {
-            $mongo->connect();
-        }
-        return $mongo->selectDB($db_name);
+    /**
+     * @param $collection
+     *
+     * @return mixed
+     */
+    private function getDb($collection)
+    {
+        $this->getConnectionInfo($collection);
+
+        return $this->mongo->selectDB($this->currentDbName);
     }
 
-    static function getCollection($collection, $read_only = false) {
-        $info = self::getConnectionInfo($collection, $read_only);
-        list($mongo, $db_name) = $info;
-        if (!$mongo->connected) {
-            $mongo->connect();
-        }
-        return $mongo->selectCollection($db_name, $collection);
+    /**
+     * @param $collection
+     *
+     * @return \MongoCollection
+     */
+    private function getCollection($collection)
+    {
+        $this->getConnectionInfo($collection);
+
+        return $this->mongo->selectCollection($this->currentDbName, $collection);
     }
 
-    static function addConnection($mongo, $db_name, $collections = null, $slave = false) {
+    /**
+     * @param      $mongo
+     * @param      $db_name
+     * @param null $collections
+     * @param bool $slave
+     */
+    public function addConnection($mongo, $db_name, $collections = null, $slave = false)
+    {
         $append = $slave ? '/slave' : '';
         if (!$collections) {
-            self::$connections[$append] = array($mongo, $db_name);
+            $this->connections[$append] = array($mongo, $db_name);
         } else {
             foreach ($collections as $c) {
-                self::$connections["$c$append"] = array($mongo, $db_name);
+                $this->connections["$c$append"] = array($mongo, $db_name);
             }
         }
     }
 
-    static function addSlaveConnection($mongo, $db_name, $collections = null) {
-        self::addConnection($mongo, $db_name, $collections, true);
+    /**
+     * @param      $mongo
+     * @param      $db_name
+     * @param null $collections
+     */
+    public function addSlaveConnection($mongo, $db_name, $collections = null)
+    {
+        $this->addConnection($mongo, $db_name, $collections, true);
     }
-    
-    static function readSlave($setting = true) {
-        self::$read_slave = $setting;
+
+    /**
+     * @param bool $setting
+     */
+    public function readSlave($setting = true)
+    {
+        $this->read_slave = $setting;
     }
-    
+
     /**
-     * Returns a MongoId from a string, MongoId, array, or Dbo object
+     * Returns a MongoId from a string, integer, MongoId, array, or Dbo object
+     *
+     * @param $obj
      *
-     * @param mixed $obj
      * @return MongoId
-     **/
-     static function id($obj) {
+     */
+    public function id($obj)
+    {
         if ($obj instanceof MongoId) {
             return $obj;
         }
+
+        if (is_int($obj)) {
```

Comment:
There's that case when you use incremental IDs in Mongo DB, using Doctrine ODM or using your own library (especially when migrating data from structured databases)
That's why I added the `is_int` check and return